### PR TITLE
Improve PolyfaceBuilder formatting and docs

### PR DESCRIPTION
**Pull request by saeeedtorabi**

### CODE REVIEW

File: core/geometry/internaldocs/figs/Polyface/polyfaceDataAfterCompress.png
Comment (saeeedtorabi) [inline core/geometry/internaldocs/figs/Polyface/polyfaceDataAfterCompress.png:1]: x coordinate of index 5 was 3. changed to 2

File: core/geometry/src/polyface/PolyfaceBuilder.ts [hunk 1]
```diff
@@ -1816,103 +1882,113 @@ export class PolyfaceBuilder extends NullGeometryHandler {
...
   public static pointsToTriangulatedPolyface(points: Point3d[], options?: StrokeOptions): IndexedPolyface | undefined {
     const graph = Triangulator.createTriangulatedGraphFromPoints(points);
     if (graph)
       return PolyfaceBuilder.graphToPolyface(graph, options);
     return undefined;
   }
-  /** Create (and add to the builder) triangles that bridge the gap between two linestrings.
-   * * Each triangle will have 1 vertex on one of the linestrings and 2 on the other
+  /**
+   * Create (and add to the builder) triangles that bridge the gap between two linestrings.
```
Comment (dassaf4) [inline core/geometry/src/polyface/PolyfaceBuilder.ts:1969]:
```suggestion
   * Add triangular facets between two linestrings.
```

File: core/geometry/src/polyface/PolyfaceBuilder.ts [hunk 2]
```diff
@@ -1816,103 +1882,113 @@ export class PolyfaceBuilder extends NullGeometryHandler {
...
+   * * Each triangle will have 1 vertex on one of the linestrings and 2 on the other.
    * * Choice of triangles is heuristic, hence does not have a unique solution.
-   * * Logic to choice among the various possible triangle orders prefers
-   *    * Make near-coplanar facets
-   *    * make facets with good aspect ratio.
+   * * Logic to choice among the various possible triangle orders prefers:
```
Comment (dassaf4) [inline core/geometry/src/polyface/PolyfaceBuilder.ts:1972]:
```suggestion
   * * Logic for choosing among the various possible triangles prefers:
```

File: core/geometry/src/polyface/PolyfaceBuilder.ts [hunk 3]
```diff
@@ -1934,17 +2010,21 @@ export class PolyfaceBuilder extends NullGeometryHandler {
...
   /**
-   * * Create (and add to the builder) quad facets for a mitered pipe that follows a centerline curve.
-   * * Circular or elliptical pipe cross sections can be specified by supplying either a radius, a pair of semi-axis lengths, or a full Arc3d.
-   *    * For semi-axis length input, x corresponds to an ellipse local axis nominally situated parallel to the xy-plane.
-   *    * The center of Arc3d input is translated to the centerline start point to act as initial cross section.
+   * Create (and add to the builder) quad facets for a mitered pipe that follows a centerline curve.
+   * * Circular or elliptical pipe cross sections can be specified by supplying either a radius, a pair of semi-axis
+   * lengths, or a full Arc3d.
+   * * For semi-axis length input, x corresponds to an ellipse local axis nominally situated parallel to the xy-plane.
+   * * The center of Arc3d input is translated to the centerline start point to act as initial cross section.
```
Comment (dassaf4) [inline core/geometry/src/polyface/PolyfaceBuilder.ts:2018]: The extra space between the asterisks is necessary to create a nested list.

```suggestion
   * Add quad facets along a mitered pipe that follows a centerline curve.
   * * Circular or elliptical pipe cross sections can be specified by supplying either a radius, a pair of semi-axis
   * lengths, or a full Arc3d:
   *    * For semi-axis length input, x corresponds to an ellipse local axis nominally situated parallel to the xy-plane.
   *    * For Arc3d input, the center is translated to the centerline start point to act as initial cross section.
```

File: core/geometry/src/test/polyface/Polyface.test.ts [hunk 4]
```diff
@@ -1672,44 +1674,38 @@ function createPolyfaceFromSynchroA(geom: any): Polyface {
...
-  // (When there are sharp edges in the mesh, this allows smaller param, point, and normal arrays.
-  //    The single-index style of Synchro and other
-  //   browser software saves memory by sharing the index array but wastes it with redundant data arrays.
-  //   The polyface compress function can get this effect)
+  // build up the index sets one facet at a time, let terminateFacet () get the facetStart table right
+  // note that polyface has separate index arrays for params and normals.
+  // (when there are sharp edges in the mesh, this allows smaller param, point, and normal arrays.
+  // the single-index style of Synchro and other
+  // browser software saves memory by sharing the index array but wastes it with redundant data arrays.
+  //  the polyface compress function can get this effect)
```
Comment (dassaf4) [inline core/geometry/src/test/polyface/Polyface.test.ts:1696]: When the comments are paragraphs, prefer proper sentence punctuation and capitalization.

```suggestion
  // Build up the index sets one facet at a time: let terminateFacet() get the facetStart table right.
  // Note that polyface has separate index arrays for params and normals (when there are sharp
  // edges in the mesh, this allows smaller param, point, and normal arrays.
  // The single-index style of Synchro and other browser software saves memory by sharing the
  // index array but wastes it with redundant data arrays.
  // The polyface compress function can remove this effect.
```

**Review by dassaf4 — APPROVED** [review]

File: core/geometry/src/polyface/PolyfaceBuilder.ts [hunk 5]
```diff
@@ -120,24 +127,25 @@ class FacetSector {
...
+   * Given two sectors with xyz and sectionDerivative (u derivative), use the edge from A to B as v direction
+   * in-surface derivative. Compute cross products (and normalize).
+   * @param sectorA "lower" sector.
+   * @param sectorB "upper" sector.
    */
   public static computeNormalsAlongRuleLine(sectorA: FacetSector, sectorB: FacetSector) {
-    // We expect that if sectionDerivative is defined so is normal.
-    // (If not, the cross product calls will generate normals that are never used ..  not good, garbage collector will clean up.)
+    // we expect that if sectionDerivative is defined so is normal (if not, the cross product calls will generate
+    // normals that are never used not good, garbage collector will clean up)
```
[the remark this answers is not in the document]
Comment (saeeedtorabi) [reply core/geometry/src/polyface/PolyfaceBuilder.ts:148]: For consistency I start all comments inside the body with lower case.

File: core/geometry/src/test/polyface/Polyface.test.ts (hunk 4 above)
Comment (saeeedtorabi) [reply core/geometry/src/test/polyface/Polyface.test.ts:1696]: For consistency I start all comments inside the body with lower case.

Comment (dassaf4) [reply core/geometry/src/test/polyface/Polyface.test.ts:1696]: This makes multi-line comments harder to read.

File: core/geometry/src/polyface/PolyfaceBuilder.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (dassaf4) [reply core/geometry/src/polyface/PolyfaceBuilder.ts:148]: That's acceptable if it's only one line.  Harder to read if multiple lines.

[the remark this answers is not in the document]
Comment (saeeedtorabi) [reply core/geometry/src/polyface/PolyfaceBuilder.ts]: ok

File: core/geometry/src/test/polyface/Polyface.test.ts (hunk 4 above)
Comment (saeeedtorabi) [reply core/geometry/src/test/polyface/Polyface.test.ts:1696]: ok

File: core/geometry/src/polyface/PolyfaceBuilder.ts [hunk 6]
```diff
@@ -59,6 +59,8 @@ import { IndexedPolyfaceSubsetVisitor } from "./IndexedPolyfaceVisitor";
 import { IndexedPolyface, PolyfaceVisitor } from "./Polyface";
 import { PolyfaceQuery } from "./PolyfaceQuery";
 
+/* eslint-disable deprecation/deprecation */
```
Comment (dassaf4) [inline core/geometry/src/polyface/PolyfaceBuilder.ts:62]: If there eslint-disable syntax that targets only the next/current line, that's a more preferable to the current syntax, which disables deprecation warnings throughout the file.

**Review by dassaf4 — APPROVED** [review]

File: core/geometry/src/polyface/PolyfaceBuilder.ts (hunk 6 above)
Comment (saeeedtorabi) [reply core/geometry/src/polyface/PolyfaceBuilder.ts:62]: I tried that. Did not work.

File: core/geometry/src/polyface/PolyfaceBuilder.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (saeeedtorabi) [reply core/geometry/src/polyface/PolyfaceBuilder.ts]: Addressed. Please resolve.

File: core/geometry/src/polyface/PolyfaceBuilder.ts (hunk 6 above)
Comment (dassaf4) [reply core/geometry/src/polyface/PolyfaceBuilder.ts:62]: You tried this line _directly above_ the recursive call on line 1227?

// eslint-disable-next-line deprecation/deprecation

Comment (saeeedtorabi) [reply core/geometry/src/polyface/PolyfaceBuilder.ts:62]: No, line 1219

image

Comment (saeeedtorabi) [reply core/geometry/src/polyface/PolyfaceBuilder.ts]: line 1227 works. Will make the change and commit

File: common/api/core-geometry.api.md [hunk 7]
```diff
@@ -4526,7 +4526,6 @@ export class PolyfaceBuilder extends NullGeometryHandler {
     addTrianglesInUncheckedConvexPolygon(ls: LineString3d, toggle: boolean): void;
     addTriangulatedRegion(region: AnyRegion): void;
     addUVGridBody(surface: UVSurface, numU: number, numV: number, uMap?: Segment1d, vMap?: Segment1d): void;
-    applyStrokeCountsToCurvePrimitives(data: AnyCurve | GeometryQuery): void;
```
Comment (dassaf4) [inline common/api/core-geometry.api.md:4529]: I don't expect the deprecated method to be removed from this file.  Other deprecated methods remain, but extract-api precedes them with `// @deprecated`.  Can you try running extract-api again?

Comment (saeeedtorabi) [reply common/api/core-geometry.api.md:4529]: Done

**Review by bbastings — APPROVED** [review]

**Pull request merged**